File: logic/test_estimate_core.php
<?php
declare(strict_types=1);

require_once __DIR__ . '/estimate_core.php';

function assert_equals($expected, $actual, string $label): void {
    if ($expected !== $actual) {
        throw new RuntimeException("FAIL: {$label} expected={$expected} actual={$actual}");
    }
    echo "OK: {$label}\n";
}

assert_equals(10000.0, calculate_estimate(new EstimateParams(10, 1000)), '10 * 1000');

try {
    calculate_estimate(new EstimateParams(0, 1000));
    throw new RuntimeException('FAIL: should throw on invalid area');
} catch (InvalidArgumentException $e) {
    echo "OK: throws on invalid area\n";
}

File: logic/test_estimate_core.py
from estimate_core import EstimateParams, calculate_estimate


def test_calculate_estimate_success():
    params = EstimateParams(area=10, unit_price=1000)
    assert calculate_estimate(params) == 10000


def test_calculate_estimate_invalid_values():
    try:
        calculate_estimate(EstimateParams(area=0, unit_price=1000))
        assert False, "should raise"
    except ValueError:
        assert True
